asf_heat_pump_suitability/analysis/exploratory/create_full_dataset/assign_cluster_suitability_and_feasibility.py

```diff
@@ -2,236 +2,402 @@
 Assigns a suitable technology and computes feasibility scoring for each cluster.
 
 This file contains functions for:
-- categorising/assigning each cluster with the most suitable technology scheme.
-- computing the feasibility scores for different heating schemes and clusters;
+- preparing the dataframe for feasibility scoring and suitability categorisation;
+- categorising/assigning each cluster with the most suitable technology.
+- computing the feasibility scores for different tech types and clusters;
 """
 
 # package imports
 import polars as pl
+import geopandas as gpd
 
+# local imports
+import config
+from asf_heat_pump_suitability.utils.save_utils import save_to_s3
 
-def assign_str_suitable_techology_scheme(
-    cluster_size: int,
-    hn_area: bool,
-    city_centre: bool,
-    has_outdoor_space: bool,
-) -> str:
+
+def prepare_df_for_suitability_categorisation(
+    df: pl.DataFrame,
+    city_centre_oas: set = config.city_centre_oas,
+    outdoor_space_threshold: float = config.outdoor_space_threshold,
+) -> pl.DataFrame:
+    """
+    Prepares the dataframe for suitability categorisation. This includes:
+    - Identifying clusters close to the city center or in heat network zones
+    - Identifying clusters with outdoor space
+
+    Args:
+        df (pl.DataFrame): DataFrame containing the dataset with feature values for each UPRN.
+        city_centre_oas (set, optional): OAs that are considered part of the city center. Should be an oa21 format.
```

Comment:
```suggestion
        city_centre_oas (set, optional): OAs that are considered part of the city center. Should be in OA21 codes format.
```
you're using the codes here right?